frontend/packages/kubevirt-plugin/src/components/cdi-upload-provider/cdi-upload-provider.tsx

```diff
@@ -0,0 +1,131 @@
+import * as React from 'react';
+import axios, { Canceler } from 'axios';
+import {
+  WatchK8sResource,
+  useK8sWatchResource,
+} from '@console/internal/components/utils/k8s-watch-hook';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { UPLOAD_STATUS } from './consts';
+import { CDIConfigModel } from '../../models';
+import { getUploadProxyURL } from '../../selectors/cdi';
+
+const resource: WatchK8sResource = {
+  kind: CDIConfigModel.kind,
+  isList: true,
```

Comment:
@rawagner I'm not sure why, but with `isList` I'm getting the route succesfully 100% of the time.
without it, there are scenarios where the route is empty, not sure why..maybe 1 out of 4 tries its empty